ts/parts/Axis.ts

```diff
@@ -5496,6 +5498,20 @@ extend(Axis.prototype, /** @lends Highcharts.Axis.prototype */{
             }
             if (!tickPositionsOption && !tickPositioner) {
                 this.adjustTickAmount();
+
+                // Make sure ticks are within axis min and max range (#12716).
+                if (this.isOrdinal) {
```

Comment:
Can we move this logic to `OrdinalAxis` file? E.g. add `afterAdjustTickAmount` event and modify config there.